feat(ui-contract-editor): updates to parsing and refactors code

DianaLease opened the pull request:
Signed-off-by: Diana Lease <dianarlease@gmail.com>

<!--- Provide a formatted commit message describing this PR in the Title above -->
<!--- See our DEVELOPERS guide below: -->
<!--- https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format -->

### Changes
<!--- More detailed and granular description of changes -->
<!--- These should likely be gathered from commit message summaries -->
- Allows editing of the clause text if that clause has a `parseable` property set to `false` on its data object
- Updates the `onClauseUpdated` function to replace the clause node if it receives a new clause node after parsing
- Adds new helper function `getClauseWithPath` and refactors previous code to use it where applicable

### Flags
<!--- Provide context or concerns a reviewer should be aware of -->
~~- If an `onClauseUpdated` method is passed to the ContractEditor component, the Contract Editor now expects it to return an array where the first element is whether or not the clause successfully parsed as a boolean and the second is an optional new node to replace the previous clause node.~~
- If an `onClauseUpdated` method is passed to the ContractEditor component, the Contract Editor now expects it to return an object. This object contains the optional properties `node`, `operation`, and `error`. This will also facilitate @dselman 's work on formulae.

irmerk commented:
I'm not understanding this flag, I don't see this in the code:
> If an `onClauseUpdated` method is passed to the ContractEditor component, the Contract Editor now expects it to return an array where the first element is whether or not the clause successfully parsed as a boolean and the second is an optional new node to replace the previous clause node.

DianaLease commented:
> I'm not understanding this flag, I don't see this in the code:
> 
> > If an `onClauseUpdated` method is passed to the ContractEditor component, the Contract Editor now expects it to return an array where the first element is whether or not the clause successfully parsed as a boolean and the second is an optional new node to replace the previous clause node.

I've now changed this logic anyway 🙃   It now expects an object. I updated the "flags"!
